test/filters/test_convolution.cpp

```diff
@@ -41,43 +41,59 @@
 #include <pcl/pcl_tests.h>
 #include <pcl/filters/convolution.h>
 #include <pcl/point_types.h>
+#include <cmath>
 
 using namespace pcl;
 using namespace pcl::test;
 
-PointCloud<PointXYZI>::Ptr input (new PointCloud<PointXYZI> ());
-Eigen::ArrayXf filter(7);
+typedef struct
+{
+  float r;
+  float g;
+  float b;
+} COLOR;
+
+static const COLOR basemap[6] = {
+  { 0.2902f, 0.0784f, 0.5490f },
+  { 0.1176f, 0.5333f, 0.8980f },
+  { 0.5451f, 0.7647f, 0.2902f },
+  { 1.0000f, 0.9216f, 0.2314f },
+  { 1.0000f, 0.0000f, 0.0000f },
+};
+
+COLOR get_color(COLOR* colormap, size_t color_count, float fmin, float fmax, float value)
+{
+  float* steps = new float[color_count];
+  float f = (fmax - fmin) / (float)(color_count - 1);
+  for (int i = 0; i < (int)color_count; i++)
+    steps[i] = fmin + f * (float)i;
+  COLOR result = { 0, 0, 0 };
+  if (value < fmin) value = fmin;
+  if (value > fmax) value = fmax;
```

Comment:
Maybe use `value = std:min(fmax, std::max(value, fmin))` instead (`std::clamp` is C++17 :-( )

Reply:
I've changed that, too.